### PR TITLE
Fix NPE in Desugar

### LINKED ISSUE #31908: NullPointerException while compiling
**Description:**
NullPointerException while compiling

**Steps to reproduce:**
* run `bal test bug-report-01.bal` (see attached `bug-report-01.bal.txt` file below)

```
Compiling source
        bug-report-01.bal
ballerina: Oh no, something really went wrong. Bad. Sad.

We appreciate it if you can report the code that broke Ballerina in 
https://github.com/ballerina-platform/ballerina-lang/issues with the 
log you get below and your sample code.

We thank you for helping make us better.

[2021-07-27 13:15:10,736] SEVERE {b7a.log.crash} - null
<stacktrace>
```

**Affected Versions:**
``` 
Ballerina Swan Lake Beta 2
Language specification 2021R1
Update Tool 1.3.5
```
**OS, DB, other environment details and versions:**
Linux integrationbox 5.8.0-63-generic #71~20.04.1-Ubuntu SMP Thu Jul 15 17:46:08 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux

**Related Issues (optional):**

**Suggested Labels (optional):**

**Suggested Assignees (optional):**

**Source file**
[bug-report-01.bal.txt](https://github.com/ballerina-platform/ballerina-lang/files/6884967/bug-report-01.bal.txt)

### PR DESCRIPTION
**Pull request by rdulmina**
Fix NPE in desugar when checked expression in let expression

Fixes  #31908

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [x] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java [hunk 1]
```diff
@@ -7732,8 +7732,8 @@ private void visitCheckAndCheckPanicExpr(BLangCheckedExpr checkedExpr, boolean i
                 getSafeAssignSuccessPattern(checkedExprVar.pos, checkedExprVar.symbol.type, true,
                                             checkedExprVar.symbol, null);
         BLangMatchTypedBindingPatternClause patternErrorCase =
-                getSafeAssignErrorPattern(checkedExpr.pos, this.env.scope.owner, checkedExpr.equivalentErrorTypeList,
-                                          isCheckPanic);
+                getSafeAssignErrorPattern(checkedExpr.pos, this.env.enclInvokable.symbol,
```
Comment (MaryamZi) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java:7735]: Still fails when outside a function.

```ballerina
float location =
        let var statusSuccess = check openFileSuccess()
        in 5.33;
```

Comment (rdulmina) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java:7736]: Fixed, thanx

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java [hunk 2]
```diff
@@ -1387,6 +1387,7 @@ public void visit(BLangSimpleVariable varNode) {
     @Override
     public void visit(BLangLetExpression letExpression) {
         SymbolEnv prevEnv = this.env;
+        letExpression.env.enclInvokable = this.env.enclInvokable;
```
Comment (pubudu91) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java:1390]: Why set this in Desugar? 🤔  Shouldn't this already be automatically set when the let expr env was created? 🤔

Comment (rdulmina) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java:1390]: It is in local level. But when in module level env.enclInvokable it sets to null

Comment (rdulmina) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java:1392]: Resolving as per the offline discussion.

**Review by MaryamZi — APPROVED** [review]

**Pull request merged**